### PR TITLE
JSDK-3018 - TypeScript Definitions for Room and top level exports

### PR DESCRIPTION
**Pull request by PikaJoyce**
JIRA Ticket : [JIRA-3018](https://issues.corp.twilio.com/browse/JSDK-3018)

This is a PR for the TypeScript definitions for the Room and other top level exports including all of their dependent

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [x] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: tsdef/error.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { TwilioError } from './TwilioError';
+
+export class MediaConnectionError extends TwilioError {
```
Comment (manjeshbhargav) [inline tsdef/error.d.ts:3]:
```suggestion
export interface MediaConnectionError extends TwilioError {
```

File: tsdef/error.d.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  code: 53405;
+  message: 'Media connection failed';
+}
+
+export class SignalingConnectionDisconnectedError extends TwilioError {
```
Comment (manjeshbhargav) [inline tsdef/error.d.ts:8]:
```suggestion
export interface SignalingConnectionDisconnectedError extends TwilioError {
```

File: tsdef/twilio-video.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { ConnectOptions, CreateLocalTrackOptions, CreateLocalTracksOptions, LocalTrack } from './types';
```
Comment (manjeshbhargav) [inline tsdef/twilio-video.d.ts:1]: Why is this file necessary?

File: tsdef/twilio-video.d.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import { LocalVideoTrack } from './LocalVideoTrack';
+import { Room } from './Room';
+
+export const isSupported: boolean;
+export const version:string;
+export function connect(token: string, options?: ConnectOptions): Promise<Room>;
+export function createLocalAudioTrack(options?: CreateLocalTrackOptions): Promise<LocalAudioTrack>;
+export function createLocalTracks(options?: CreateLocalTracksOptions): Promise<LocalTrack[]>;
+export function createLocalVideoTrack(options?: CreateLocalTrackOptions): Promise<LocalVideoTrack>;
+export function rewriteLocalTrackIds(room: Room, trackStats: LocalTrackStats[]): LocalTrackStats[];
```
Comment (manjeshbhargav) [inline tsdef/twilio-video.d.ts:13]:
```suggestion
```

File: tsdef/types.d.ts [hunk 5]
```diff
@@ -68,3 +70,87 @@ export interface NetworkQualityConfiguration {
...
+  enableDscp?: boolean;
+  iceServers?: RTCIceServer[];
+  iceServersTimeout?: number;
+  iceTransportPolicy?: RTCIceTransportPolicy;
+  insights?: boolean;
+  maxAudioBitrate?: number | null;
+  maxVideoBitrate?: number | null;
+  name?: string | null;
+  networkQuality?: boolean | NetworkQualityConfiguration;
+  region?: 'au1' | 'br1' | 'ie1' | 'de1' | 'jp1' | 'sg1' | 'us1' | 'us2' | 'gll';
```
Comment (manjeshbhargav) [inline tsdef/types.d.ts:139]:
```suggestion
  region?: string;
```

**Review by manjeshbhargav — CHANGES_REQUESTED** [review]

File: tsdef/twilio-video.d.ts (hunk 3 above)
Comment (PikaJoyce) [reply tsdef/twilio-video.d.ts:1]: I included these imports since they're part of the top level exports options.

File: tsdef/TwilioError.d.ts [hunk 6]
```diff
@@ -4,3 +4,13 @@ export class TwilioError extends Error {
...
+
+export interface MediaConnectionError extends TwilioError {
+  code: 53405;
+  message: 'Media connection failed';
+}
+
+export interface SignalingConnectionDisconnectedError extends TwilioError {
+  code: 53001;
+  message: 'Signaling connection disconnected';
+}
```
Comment (manjeshbhargav) [inline tsdef/TwilioError.d.ts:16]: @PikaJoyce Let's leave out the specific errors for now. We'll add them later.

**Review by manjeshbhargav — APPROVED** [review]
